lib/galaxy/datatypes/genetics.py

```diff
@@ -827,6 +829,297 @@ def __init__(self, **kwd):
                                 substitute_name_with_metadata='base_name', is_binary=True)
 
 
+class LinkageStudies(Text):
+    """
+    superclass for classical linkage analysis suites
+    """
+    test_files = [
+        'linkstudies.allegro_fparam', 'linkstudies.alohomora_gts',
+        'linkstudies.linkage_datain', 'linkstudies.linkage_map'
+    ]
+
+    def __init__(self, **kwd):
+        Text.__init__(self, **kwd)
+        self.lcount = 0
+        self.max_lines = 10
+        # iterated whole file without errors
+        self.eof_res = True
+
+    def header_check(self, fio):
+        """
+        Overrideable post-binary file check function
+        """
+        return True
+
+
+class GenotypeMatrix(LinkageStudies):
+    """
+    Sample matrix of genotypes
+    - GTs as columns
+    """
+    file_ext = "alohomora_gts"
+
+    def __init__(self, **kwd):
+        super(GenotypeMatrix, self).__init__(**kwd)
+        self.num_cols = -1
+
+    def header_check(self, fio):
+        header_elems = fio.readline().split('\t')
+
+        if header_elems[0] != "Name":
+            return False
+
+        try:
+            return all([int(sid) > 0 for sid in header_elems[1:]])
+        except ValueError:
+            return False
+
+        return True
+
+    def sniff(self, filename):
+        """
+        >>> classname = GenotypeMatrix
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> extn_true = classname().file_ext
+        >>> file_true = get_test_fname("linkstudies." + extn_true)
+        >>> classname().sniff(file_true)
+        True
+
+        >>> false_files = list(LinkageStudies.test_files)
+        >>> false_files.remove("linkstudies." + extn_true)
+        >>> result_true = []
+        >>> for fname in false_files:
+        ...     file_false = get_test_fname(fname)
+        ...     res = classname().sniff(file_false)
+        ...     if res:
+        ...         result_true.append(fname)
+        >>>
+        >>> result_true
+        []
+        """
+        with open(filename, "r") as fio:
+
+            if not self.header_check(fio):
+                return False
+
+            for line in fio:
```

Comment:
use something like

```python
for lcount, line in enumerate(fio)
    if lcount > ...
```

instead and then you can remove self.lcount everywhere. It seems strange since it isn't the real lcount, it's "the number of lines I've seen so far as long as it isn't greater than 10". I would find myself mis-using that variable if I had not read the code closely enough.